eventcrypto_test.go

```diff
@@ -258,3 +261,127 @@ func TestSignEventTestVectors(t *testing.T) {
    	 	}
 	}`)
 }
+
+type StubVerifier struct {
+	requests []VerifyJSONRequest
+	results  []VerifyJSONResult
+}
+
+func (v *StubVerifier) VerifyJSONs(ctx context.Context, requests []VerifyJSONRequest) ([]VerifyJSONResult, error) {
+	v.requests = append(v.requests, requests...)
+	return v.results, nil
+}
+
+func TestVerifyEventSignatures(t *testing.T) {
+	verifier := StubVerifier{}
+
+	eventJSON := []byte(`{
+		"type": "m.room.name",
+		"state_key": "",
+		"event_id": "$test:localhost",
+		"room_id": "!test:localhost",
+		"sender": "@test:localhost",
+		"origin": "originserver",
+		"content": {
+			"name": "Hello World"
+		},
+		"origin_server_ts": 123456
+	}`)
+
+	var event Event
+	if err := json.Unmarshal(eventJSON, &event.fields); err != nil {
+		t.Fatal(err)
+	}
+	event.eventJSON = eventJSON
+
+	events := []Event{event}
+	if err := VerifyEventSignatures(context.Background(), events, &verifier); err != nil {
+		t.Fatal(err)
+	}
+
+	// should be two call to the verifier
```

Comment:
Should this be "There should be two verification requests".
(I find the use of "call" confusing because VerifyJSONs is only called once)